docs/src/pages/customization/components/components.md

```diff
@@ -319,3 +319,41 @@ const theme = createMuiTheme({
 ```
 
 {{"demo": "pages/customization/components/GlobalThemeOverride.js"}}
+
+### Adding new component variants
+
+You can take advantage of the `variants` key of the `theme` to add new variants to Material-UI components.
```

Comment:
I think this section should explain more what this is doing immediately followed by the visual example. The TypeScript section should come last. At least I don't care about API specifics before I know what problem it is solving. 

For reference [rendered preview](https://deploy-preview-21648--material-ui.netlify.app/customization/components/#adding-new-component-variants)

Reply:
Expanded, let me know if you think more details are needed.

Reply:
That's better. It's still quite a bit to take in but we'll have to see.